### ISSUE TITLE
OpenAI and Azure OpenAI - calls one after another

### ISSUE THREAD

**Issue by ushakrishnan** [langchain-ai/langchain#5537]
### System Info

langchain - 0.0.174 / 0.0.178 / 0.0.187
python3

### Who can help?

@hwchase17 @agola11 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [X] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

1. Call OpenAI setting the parameters -  openai_api_type , openai_api_version , openai_api_base , openai_api_key
-- Successful OpenAI request response
2. Call Azure OpenAI setting the parameters -  openai_api_type , openai_api_version , openai_api_base , openai_api_key
-- Fails
All subsequent calls fails

Alternatively, if you first call Azure OpenAI with parameters set correctly - that succeeds, but OpenAI fails. And all subsequent fails.

Each independently works - so guessing the. parameter values work as expected.  But when one after other is called, the second API (OpenAI or Azure OpenAI - which ever is called second) - fails

### Expected behavior

If parameters are set correctly, both should work as required.  If independently they work if app is restarted, why would it fail if they are called sequentiially?

**Comment by Xmaster6y** [comment]
I think it is because the variables were set globally (using `openai.api_typ=...`) at the embedding object definition. It has been changed in v0.0.192. Could you check if it solves the issue? And if not, can you provide a sample of code to reproduce the issue.

**Comment by ushakrishnan** [comment]
@Xmaster6y  

NOPE!  Not fixed yet!!  Tried with version v0.0.195 (current).

Here are the two pages
- https://github.com/ushakrishnan/SearchWithOpenAI/blob/main/pages/230_Talk_with_Azure_Open_AI.py
- https://github.com/ushakrishnan/SearchWithOpenAI/blob/main/pages/130_Talk_with_Open_AI.py

Both works independently, when not called after each other.  Irrespective of which is called first, the OpenAI is set to be that - for ex. calling Azure OpenAI, then expects all following to have variables for Azure OpenAI - even after resetting environment variables.  Same is the case in the other order called. 

So, for now, I have hidden the pages once the user makes a choice to avoid errors.

**Comment by Xmaster6y** [comment]
I think it's because you shouldn't setup openai yourself since langchain already take care of this.

The two following samples of code work for me (v0.0.198).

```python
llm = OpenAI(
    model_name="text-davinci-003",
    temperature=0.0,
    max_tokens=10,
    openai_organization=openai_org, 
    openai_api_key=openai_key
)
print(llm("Hello who are you?"))
llm_az = AzureOpenAI(
    deployment_name="text-davinci-003",
    openai_api_version="2023-05-15",
    temperature=0.0,
    max_tokens=10,
    openai_api_base=azure_url, 
    openai_api_key=azure_key
)
print(llm_az("Hello who are you?"))
```
```python
llm = OpenAI(
    model_name="text-davinci-003",
    temperature=0.0,
    max_tokens=10,
    openai_organization=openai_org, 
    openai_api_key=openai_key
)
llm_az = AzureOpenAI(
    deployment_name="text-davinci-003",
    openai_api_version="2023-05-15",
    temperature=0.0,
    max_tokens=10,
    openai_api_base=azure_url, 
    openai_api_key=azure_key
)
print(llm("Hello who are you?"))
print(llm_az("Hello who are you?"))
```
